### ISSUE TITLE
Official RPi Touchscreen rpi-f5406 swapxy

### ISSUE THREAD

**Issue by PapaSierraRomeo** [raspberrypi/firmware#537]
Hi,

I'm using the official RPi Touchscreen in portrait mode but am having problems getting the touchscreen to work in this orientation. My application, which is a Python/pygame app run from the command line,  does not need X Windows so the common means by which the touchscreen axes can be swapped (xinput set-prop 'FT5406 memory based driver' 'Evdev Axes Swap' 1 etc) won't help. I notice that touchscreen drivers like the ads7846 offer the ability to swapxy axes but the ft5406 driver has no similar parameters. 

Is there a way to achieve this with the rpi-ft5406 driver without using X-Windows? Would a swapxy parameter be a useful addition to the code?

Many thanks.

**Comment by popcornmix** [comment]
I assume you can swap X/Y inside your game?
Presumably you are having to render your graphics to the screen with swapped X/Y and using swapped width/height from the framebuffer?

**Comment by PapaSierraRomeo** [comment]
Thanks for responding. Swapping x and y in the game leaves a section of the screen which cannot be accessed owing to the 800x480 proportion so only about half of the screen is usable.

For rendering out the game I can use the display_rotate=1 command in /boot/config.txt which sorts out the display correctly and means that I don’t have to do anything special from my code.

**Comment by Ruffio** [comment]
@PapaSierraRomeo so your issue has been solved?

**Comment by PapaSierraRomeo** [comment]
Not really. In the end I have gone the X-Windows route and am using the well established methods of rotating the touchscreen interface under X. Before that, and to try to maintain the command line only method of operation, I tried scaling and swapping the axes. This sort of worked but relied on the first touch of the screen when the program started being at 0,0 and only then would the correct co-ordinates be returned for subsequent position calls when other areas of the screen were touched. I tried to find a way to initialise the screen but couldn’t so ended up going the X-Windows route instead as forcing (or relying) on the user to initialise the display didn’t seem a viable solution.
